NAS-100121: Begin Replication/SSH Connections/SSH Keypairs/Periodic Snapshot tasks rewrite (#1167)
freenas/freenas-docs: userguide/storage.rst

   |                   |                   |                                                                                            |
   +-------------------+-------------------+--------------------------------------------------------------------------------------------+
   | Recursive         | checkbox          | Set to take separate snapshots of the volume or dataset and each of its child datasets.    |
   |                   |                   | Unset to take a single snapshot of only the specified volume or dataset.                   |
   |                   |                   |                                                                                            |
   +-------------------+-------------------+--------------------------------------------------------------------------------------------+
   | Exclude           | string            | Exclude specific child datasets from the snapshot. Use with :guilabel:`Recursive`          |
   |                   |                   | snapshots. List paths to any child datasets to exclude. Example:                           |

>    |                   |                   | :samp:`pool1/dataset1/child1`. A recursive snapshot of :file:`pool1/dataset1` will include |

Reviewer: ```suggestion
   |                   |                   | :samp:`pool1/dataset1/child1`. A recursive snapshot of :file:`pool1/dataset1` includes     |
```

   |                   |                   | year, month, day of month, hour, and minute as defined in                                  |
   |                   |                   | `strftime(3) <https://www.freebsd.org/cgi/man.cgi?query=strftime>`__. Example:             |
   |                   |                   | :literal:`backups_%Y-%m-%d_%H:%M`                                                          |
   +-------------------+-------------------+--------------------------------------------------------------------------------------------+
   | Begin             | drop-down menu    | Hour and minute when the system can begin taking snapshots.                                |
   |                   |                   |                                                                                            |
   +-------------------+-------------------+--------------------------------------------------------------------------------------------+
   | End               | drop-down menu    | Hour and minute the system must stop creating snapshots. Snapshots already in progress     |

>    |                   |                   | will continue until complete                                                               |

Reviewer: ```suggestion
   |                   |                   | continue until complete                                                                    |
```

replication task is complete.

Later replications only send incremental snapshot changes to the
destination system. This reduces both the total space required by
replicated data and the network bandwidth required for the replication
to complete.

The target dataset on the destination system is created in *read-only*
mode to protect the data. To mount or browse the data on the destination
system, use a clone of the snapshot. Clones are created in *read/write*
mode, making it possible to browse or mount them. See :ref:`Snapshots`
for more details.

Examples in this section refer to the %brand% system with the original
datasets for snapshot and replication as |rpln-sys1| and the %brand%
system that is storing replicated snapshots as |rpln-sys2|.

.. index:: Add Replication
.. _Add Replication:

Add Replication
~~~~~~~~~~~~~~~

To create a new replication, go to
:menuselection:`Storage --> Replication Tasks`
and click :guilabel:`Add Replication`.

.. _storage_replication_add_fig:

.. figure:: images/storage-replication-add.png

   Create a New Replication

Creating remote, local, %brand% legacy engine, and one-time replications

> are all possible with this screen.

Reviewer: ```suggestion
are all possible through this screen.
```

   +---------------------------+-----------+----------------+-----------------------------------------------------------------------------------------------------------------+
   | Enabled                   | ALL       | checkbox       | Activates the replication schedule.                                                                             |
   +---------------------------+-----------+----------------+-----------------------------------------------------------------------------------------------------------------+

Saving a new replication adds an entry to
:menuselection:`Storage --> Replication Tasks`.
The columns show the various settings for the replication. The
:guilabel:`State` shows if the replication has run successfully or if
an error has occurred.

.. _zfs_repl_task_list_fig:

.. figure:: images/storage-replication-tasks.png
   :width: 90%

   Replication Task List

To see more options for a saved replication, click |ui-options| for that
task. There are options to :guilabel:`Delete`, :guilabel:`Edit`, or
immediately start that replication.

> When multiple replications have the same schedule, they will run

Reviewer: ```suggestion
When multiple replications have the same schedule, they run
```

   new replicated data can be created.

.. _Limiting Replication Times:

Limiting Replication Times
~~~~~~~~~~~~~~~~~~~~~~~~~~

The :guilabel:`Schedule`, :guilabel:`Begin`, and :guilabel:`End` times
in a replication task make it possible to restrict when replication is
allowed. These times can be set to only allow replication after business

> hours, or at other times when disk or network activity will not slow

Reviewer: ```suggestion
hours, or at other times when disk or network activity does not slow
```
